packages/upload/src/FzUpload.vue

```diff
@@ -97,7 +116,13 @@ function handleInputChange(event: Event) {
 }
 
 function addFiles(filesToAdd: File[]) {
-  model.value = [...(model.value ?? []), ...filesToAdd];
+  if (props.multiple) {
+    model.value = [...(model.value ?? []), ...filesToAdd];
+  } else {
+    model.value = [filesToAdd[0]];
+  }
+
+  emit("fzupload:change", model.value);
```

Comment:
this works as is? I thought changing a model was async

Reply:
it should works, but to be safe I changed it